core/src/main/java/org/jruby/RubyString.java

```diff
@@ -1486,9 +1487,18 @@ public IRubyObject initialize19(ThreadContext context) {
         return this;
     }
 
-    @JRubyMethod(name = "initialize", visibility = PRIVATE)
-    public IRubyObject initialize19(ThreadContext context, IRubyObject arg0) {
-        replace19(arg0);
+    @JRubyMethod(name = "initialize", visibility = PRIVATE, optional = 2)
+    public IRubyObject initialize19(ThreadContext context, IRubyObject[] args) {
+        final String[] validArgs = new String[]{"encoding"};
```

Comment:
I'm not quite sure what Java will do here - since it's a final String[], will it be optimized away, or is this going to result in an allocation per invocation? Should this be moved to an array initialized outside of the method?